File: lab-1/src/main/java/com/evist0/labforwardlist/factory/impl/StringFactory.java
package com.evist0.labforwardlist.factory.impl;

import com.evist0.labforwardlist.factory.ObjectFactory;

import java.util.Comparator;

public class StringFactory implements ObjectFactory {
    @Override
    public Object parse(String input) {
        return input;
    }

    @Override
    public String getObjectName() {
        return "String";
    }

    @Override
    public Comparator getComparator() {
        return Comparator.comparing(o -> ((String) o));
    }
}

File: lab-2/src/main/scala/com/hamaddo/labforwardlist/factory/impl/StringFactory.scala
package com.hamaddo.labforwardlist.factory.impl

import com.hamaddo.labforwardlist.factory.ObjectFactory

import java.util.Comparator

private class StringFactory extends ObjectFactory {
  override def parse(input: String): String = {
    input
  }

  override def getObjectName: String = {
    "String"
  }

  override def getComparator: Comparator[Object] = {
    Comparator.comparing(o => o.asInstanceOf[String])
  }
}
